backend/src/main/java/com/htecgroup/skynest/service/impl/FileServiceImpl.java

```diff
@@ -139,6 +141,47 @@ public List<FileResponse> getAllFilesWithParent(UUID parentFolderId) {
     return asFileResponseList(allFiles);
   }
 
+  @Override
+  public void moveFileToFolder(UUID fileId, UUID destinationFolderId) {
+    FileMetadataEntity fileMetadataEntity = findFileMetaDataEntity(fileId);
+    FolderEntity folderEntity =
+        folderRepository.findById(destinationFolderId).orElseThrow(FolderNotFoundException::new);
+    checkIfFileAlreadyInsideFolder(fileMetadataEntity, folderEntity);
+    fileMetadataEntity.setParentFolder(folderEntity);
+    saveMoveFile(fileMetadataEntity);
+  }
+
+  @Override
+  public void moveFileToRoot(UUID fileId) {
+    FileMetadataEntity fileMetadataEntity = findFileMetaDataEntity(fileId);
+    checkIfFileIsAlreadyInsideRoot(fileMetadataEntity);
+    fileMetadataEntity.setParentFolder(null);
```

Comment:
you can create a method in FileMetadataEntity and name it for example moveToRoot() and move this one line there. I believe you already did something simular with the deleteBucket/Folder functionality